upgrade.php

```diff
@@ -353,13 +374,32 @@
     $composer_update = shell_exec('php composer.phar self-update');
     echo $composer_update."\n\n";
 
+
+
+    // Use --no-dev only if we are in production mode.
+    // This will cause errors otherwise, if the user is in develop or local for their APP_ENV
+    if ($app_environment == 'production') {
```

Comment:
If you wanted to, you might be able to fill up a `$composer` variable with either `php composer.phar` or `composer` (based on what's there) and then you might be able to cut down a little on some of this duplication. Minor issue though, not important.

Reply:
Yeah, I know, but it's late and I'm hungry, and this is better than what we had.